### PR TITLE
NetSim: Teacher View

### PR DESCRIPTION
**Pull request by islemaster**
Adds a "Teacher View" that is only available from the lobby, and only to shard owners (i.e. the teacher).

Here's the button:
screenshot from 2016-07-29 16-01-01

It's just like the log browser, but it includes a "Sent By" column listing the username for each message.
screenshot from 2016-07-29 16-02-42

It also lets the teacher filter by "Sent By" user, which makes it easy to track a student across simulated IP addresses.
screenshot from 2016-07-29 16-02-54

I've intentionally kept this separate from the main log browser view, which teachers may still want to project (say, during the battleship lesson) for instructional purposes.  We also enable the "Teacher View" button in levels that don't have the log browser (like Battleship) to allow monitoring.
# TODO
- [x] Disable Teacher View on peer-to-peer levels, or else add logging in those levels!

### CODE REVIEW

File: apps/src/netsim/NetSimLogBrowser.jsx [hunk 1]
```diff
@@ -179,6 +204,26 @@ if (BUILD_STYLEGUIDE) {
...
                 />
               </div>
-            ));
+            ))
+      .addWithInfo(
+        `Teacher's View`,
+        `Here's what the teacher (or shard owner) gets to see`,
+        () => (
+          <div id="netsim">
+            <div className="new-router-log-modal">
```
Comment (islemaster) [inline apps/src/netsim/NetSimLogBrowser.jsx]: TODO: Remove now-unneeded wrapper div

File: apps/src/netsim/NetSimLogBrowserFilters.jsx [hunk 2]
```diff
@@ -123,3 +135,38 @@ const TrafficFilterDropdown = React.createClass({
...
+        <option value="none">
+          {this.props.i18n.logBrowserHeader_sentByAnyone()}
+        </option>
+        {_.uniq(this.props.logRows.map(row => row.sourceUserName))
+          .sort((a, b) => a.localeCompare(b))
+          .map(name => (
+          <option value={`by ${name}`} key={name}>
+            {this.props.i18n.logBrowserHeader_sentByName({name})}
+          </option>
+        ))}
```
Comment (islemaster) [inline apps/src/netsim/NetSimLogBrowserFilters.jsx]: TODO: Indentation here looks wrong.

File: apps/src/netsim/NetSimLogBrowserTable.jsx [hunk 3]
```diff
@@ -83,36 +85,49 @@ const NetSimLogBrowserTable = React.createClass({
...
+    if (this.props.teacherView) {
+      columns.push({
         header: {
-          label: 'Logged By',
+          label: 'Sent By',
           transforms: [sortable],
           props: {style: style.nowrap}
         },
-        cell: {property: 'logged-by', props: {style: style.nowrapTd}}
+        cell: {property: 'sourceUserName', props: {style: style.nowrapTd}}
```
Comment (islemaster) [inline apps/src/netsim/NetSimLogBrowserTable.jsx]: TODO: Rename sourceUserName/sent-by

**Comment by Hamms** [comment]
LGTM!

**Pull request merged**